direct/io-bulkfs-parquet/src/test/java/cz/o2/proxima/direct/bulk/fs/parquet/StreamElementWriteSupportTest.java

```diff
@@ -0,0 +1,441 @@
+
+    // Delete should skip attribute value
+    recordConsumer.clear();
+    writer.write(StreamElement.delete(entity, attr, uuid, "key", attr.getName(), now));
+    assertEquals(
+        String.format(
+            "<message>"
+                + "<attribute_prefix[5]>data</attribute_prefix[5]>"
+                + "<attribute[4]>data</attribute[4]>"
+                + "<uuid[1]>%s</uuid[1]>"
+                + "<operation[3]>d</operation[3]>"
+                + "<key[0]>key</key[0]>"
+                + "<timestamp[2]>%d</timestamp[2]>"
+                + "</message>",
+            uuid, now),
+        recordConsumer.getWritten());
+    // Same for delete wildcard
+
+    recordConsumer.clear();
+    writer.write(
+        StreamElement.deleteWildcard(entity, entity.getAttribute("wildcard.*"), uuid, "key", now));
+    assertEquals(
+        String.format(
+            "<message>"
+                + "<attribute_prefix[5]>wildcard.*</attribute_prefix[5]>"
+                + "<attribute[4]>wildcard.*</attribute[4]>"
+                + "<uuid[1]>%s</uuid[1]>"
+                + "<operation[3]>w</operation[3]>"
+                + "<key[0]>key</key[0]>"
+                + "<timestamp[2]>%d</timestamp[2]>"
+                + "</message>",
+            uuid, now),
+        recordConsumer.getWritten());
+
+    // Check upsert for wildcard
+
+    recordConsumer.clear();
+    writer.write(
+        StreamElement.upsert(
+            entity,
+            entity.getAttribute("wildcard.*"),
+            uuid,
+            "key",
+            "key.123",
+            now,
+            "value".getBytes(StandardCharsets.UTF_8)));
+
+    assertEquals(
+        String.format(
+            "<message>"
+                + "<attribute_prefix[5]>wildcard.*</attribute_prefix[5]>"
+                + "<attribute[4]>key.123</attribute[4]>"
+                + "<uuid[1]>%s</uuid[1]>"
+                + "<operation[3]>u</operation[3]>"
+                + "<key[0]>key</key[0]>"
+                + "<timestamp[2]>%d</timestamp[2]>"
+                + "<wildcard[7]>value</wildcard[7]>"
+                + "</message>",
+            uuid, now),
+        recordConsumer.getWritten());
+  }
+
+  private <T> void assertWritten(
+      String expected,
+      SchemaTypeDescriptor<T> descriptor,
+      String attributeName,
+      int index,
+      T value) {
+    @SuppressWarnings("unchecked")
+    final Writer<T> writer =
+        (Writer<T>)
+            writeSupport.createWriter(
+                descriptor,
+                attributeName,
+                new GroupType(
+                    Repetition.REQUIRED,
+                    "test-schema",
+                    ParquetUtils.mapSchemaTypeToParquet(descriptor, attributeName)));
+    writer.setName(attributeName);
+    writer.setIndex(index);
+    writer.write(value);
+    assertEquals(expected, recordConsumer.getWritten());
+  }
+
+  private static class TestRecordConsumer extends RecordConsumer {
```

Comment:
Either this or `cz.o2.proxima.direct.bulk.fs.parquet.TestRecordConsumer` should be removed

Reply:
Thanks, I forgot this during refactoring.